### PR TITLE
fixes SPIN-1749: get package name when no jenkins trigger present.

### PR DESCRIPTION
**Pull request by zanthrash**
The quip stage needs to get the package name to patch the instacne.  When using a Jenkins Job it is able pull this off the pipelines trigger object.

However Quip is often used with a Jenkins stage instead of a trigger resulting in an empty version resolving.

This fix also looks at both the buildInfo on the context and trigger to get the target deb version and then build the version name.

@dzapata @ajordens @robfletcher @cfieber PTAL

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
ping

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy [hunk 1]
```diff
@@ -55,11 +57,48 @@ class PackageInfoSpec extends Specification {
       [["fileName": "test-package_1.0.0.deb"]]    | "test-package"                                 | "test-package_1.0.0"
       [["fileName": "test-package_1.0.0.deb"]]    | "another-package"                              | "another-package"
       [["fileName": "test-package_1.0.0.deb"]]    | "another-package test-package"                 | "another-package test-package_1.0.0"
-
+      [["fileName": "test-package_1.0.0.deb"]]    | "ssh://git@stash.corp.netflix.com:7999/uiplatform/nodequark.git?v1.0.51"| "ssh://git@stash.corp.netflix.com:7999/uiplatform/nodequark.git?v1.0.51"
```
Comment (dzapata) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: What's the behavior if the package is a url?  I didn't know we supported that.

**Comment by dzapata** [comment]
LGTM

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy (hunk 1 above)
Comment (zanthrash) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy]: if it is a url we just return from the createAugmentedRequest method

`    if (isUrl(request.package)) {
      return request
    }`

**Pull request merged**